packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -177,44 +450,57 @@ const Dropdown = forwardRef(
     }, [size, rtl, insideOverflowContainer, insideOverflowWithTransformContainer, extraStyles, multi, multiline]);
 
     const Menu = useCallback(
-      props => (
+      (props: CustomMenuProps) => (
         <MenuComponent {...props} Renderer={menuRenderer} dropdownMenuWrapperClassName={dropdownMenuWrapperClassName} />
       ),
       [dropdownMenuWrapperClassName, menuRenderer]
     );
 
-    const DropdownIndicator = useCallback(props => <DropdownIndicatorComponent {...props} size={size} />, [size]);
+    const DropdownIndicator = useCallback(
+      (props: React.HTMLAttributes<HTMLElement> & { size?: SIZES_VALUES }) => (
+        <DropdownIndicatorComponent {...props} size={size} />
+      ),
+      [size]
+    );
 
     const Option = useCallback(
-      props => (
+      (props: CustomOptionProps) => (
         <OptionComponent {...props} Renderer={finalOptionRenderer} optionWrapperClassName={optionWrapperClassName} />
       ),
       [finalOptionRenderer, optionWrapperClassName]
     );
 
-    const Input = useCallback(props => <components.Input {...props} aria-label="Dropdown input" />, []);
+    const Input = useCallback((props: InputProps) => <components.Input {...props} aria-label="Dropdown input" />, []);
 
     const SingleValue = useCallback(
-      props => (
+      (props: CustomSingleValueProps) => (
         <SingleValueComponent
           {...props}
           readOnly={readOnly}
           Renderer={finalValueRenderer}
-          selectedOption={selectedOptions[0]}
+          selectedOption={(selectedOptions as DropdownDefaultValue[])[0]}
           singleValueWrapperClassName={singleValueWrapperClassName}
         />
       ),
       [finalValueRenderer, readOnly, selectedOptions, singleValueWrapperClassName]
     );
 
-    const ClearIndicator = useCallback(props => <ClearIndicatorComponent {...props} size={size} />, [size]);
+    const ClearIndicator = useCallback(
+      (props: React.HTMLAttributes<HTMLElement> & { size?: SIZES_VALUES }) => (
+        <ClearIndicatorComponent {...props} size={size} />
+      ),
+      [size]
+    );
 
     const onOptionRemove = useMemo(() => {
-      return function (optionValue, e) {
+      return function (optionValue: number, e: React.MouseEvent | React.KeyboardEvent) {
         if (customOnOptionRemove) {
-          customOnOptionRemove(selectedOptionsMap[optionValue]);
+          customOnOptionRemove((selectedOptionsMap as unknown as DropdownDefaultValue[])[optionValue]);
         }
-        const newSelectedOptions = selectedOptions.filter(option => option.value !== optionValue);
+        const newSelectedOptions = Array.isArray(selectedOptions)
```

Comment:
Is this because of typing issues? Isn't it always an array?

Reply:
there is a default value that is not an array, so I added this to check.
and yes there was a typing issue.